### PR TITLE
Facilitator init optimization and bug fixes

### LINKED ISSUE #212: Facilitator init & start compatibility for Gateway Config
### FACILITATOR INIT

- To support token init `--gateway-config` argument is added in facilitator init.

- `--gateway-config` argument is optional when initializing facilitator for baseToken.

- `--mosaic-config` argument is optional when initializing facilitator for EIP20 tokens.

- Make sure proper validations are in place based on argument `--token-config`.

- Proposed command:
```
./facilitator init --gateway-config <gateway-config-path> --aux-chain-id <aux-chain-id> --origin-password <origin-password> --auxiliary-password <auxiliary-password> --origin-rpc <origin-rpc> --auxiliary-rpc <auxiliary-rpc> --origin-graph-ws <origin-graph-ws> --origin-graph-rpc <origin-graph-rpc> --auxiliary-graph-ws <auxiliary-graph-ws> --auxiliary-graph-rpc <auxiliary-graph-rpc> --db-path <db-path> --force
```

- Update readme.

### FACILITATOR START
- To support facilitator start for token "--token-config" argument is added.

- `--gateway-config` argument is optional when starting facilitator for baseToken.

- `--mosaic-config` argument is optional when starting facilitator for EIP20 tokens

- Make sure proper validations are in place based on argument `--gateway-config`.

- Proposed command:
```
./facilitator start --facilitator-config <facilitator-config> --gateway-config <gateway-config>
```
- Validate seed data with mosaic/gateway config.

- Update readme.

### PR DESCRIPTION
**Pull request by gulshanvasnani**
PR contains below optimization for facilitator init :
1. Changed datatype of auxchain id from string to number consistent.
2. Changed datatype of auxchain id in faciltiator-schema to number.
3. Updated test-cases to reflect the change.
4. Loading of gatewayconfig when gateway config option is specified.

fixes #212

### CODE REVIEW

**Comment by sarvesh-ost** [comment]
Could you please specify in the PR description about bug and optimisation you have done.

File: test_integration/05_facilitator_stop/01_stop.test.ts [hunk 1]
```diff
@@ -12,6 +12,6 @@ describe('facilitator stop', async (): Promise<void> => {
       facilitatorKill,
       { stdio: [process.stdout, process.stderr], env: process.env },
     );
-    fs.removeSync(Directory.getFacilitatorConfigPath(Constants.auxChainId.toString()));
+    fs.removeSync(Directory.getFacilitatorConfigPath(parseInt(Constants.auxChainId)));
```
Comment (abhayks1) [inline test_integration/05_facilitator_stop/01_stop.test.ts]: We should update Constants.auxChainId to be number. It's string now "1000".
After that we don't need to parseInt. Wherever this constants is called parseInt or parse to Number can be removed.

File: test/FacilitatorConfig/remove.test.ts [hunk 2]
```diff
@@ -24,7 +24,7 @@ import SpyAssert from '../test_utils/SpyAssert';
 
 describe('FacilitatorConfig.remove()', () => {
   it('should remove facilitator config from default path', async () => {
-    const someChain = 'someChain';
+    const someChain = 300;
```
Comment (abhayks1) [inline test/FacilitatorConfig/remove.test.ts]:
```suggestion
    const someAuxChainId = 300;
```

File: test/Database/create.test.ts [hunk 3]
```diff
@@ -24,15 +24,16 @@ import Directory from '../../src/Directory';
 import assert from '../test_utils/assert';
 import SpyAssert from '../test_utils/SpyAssert';
 
-const chainId = '1';
+const chainId = 1;
```
Comment (abhayks1) [inline test/Database/create.test.ts]:
```suggestion
const auxChainId = 1;
```

File: src/bin/facilitator-init.ts [hunk 4]
```diff
@@ -133,10 +133,18 @@ commander
           {
             originChainId,
             gatewayAddresses,
-          } = FacilitatorInit.getFromMosaicConfig(auxChainId, options.mosaicConfig)
+          } = FacilitatorInit.getFromGatewayConfig(auxChainId, options.gatewayConfig)
```
Comment (abhayks1) [inline src/bin/facilitator-init.ts]: Please add unit test case for this.

Comment (gulshanvasnani) [reply src/bin/facilitator-init.ts]: As discussed, as it is bin file and it needs refactoring before we write unit test-cases. Ticket https://github.com/mosaicdao/facilitator/issues/240 has been created to refactor code for faciltiator-init binary.

**Pull request merged**